mc-oblivious-ram/src/evictor/mod.rs

```diff
@@ -409,6 +411,281 @@ where
+) where
+    ValueSize: ArrayLength<u8> + PartialDiv<U8> + PartialDiv<U64>,
+    Z: Unsigned + Mul<ValueSize> + Mul<MetaSize>,
+    Prod<Z, ValueSize>: ArrayLength<u8> + PartialDiv<U8>,
+    Prod<Z, MetaSize>: ArrayLength<u8> + PartialDiv<U8>,
+{
+    //need one more for stash.
+    let meta_len = branch.meta.len();
+    let meta_len_with_stash = meta_len + 1;
+
+    let mut deepest_meta = vec![FLOOR_INDEX; meta_len_with_stash];
+    let mut target_meta = vec![FLOOR_INDEX; meta_len_with_stash];
+    prepare_deepest::<ValueSize, Z>(&mut deepest_meta, stash_meta, &branch.meta, branch.leaf);
+    prepare_target::<ValueSize, Z>(&mut target_meta, &mut deepest_meta, &branch.meta);
+
+    // Initializing the held data and held meta with some default values.
+    let mut dummy_held_data: A64Bytes<ValueSize> = Default::default();
+    let mut dummy_held_meta: A8Bytes<MetaSize> = Default::default();
+
+    // Held data and held meta represent the elements we have picked up iterating
+    // from the stash to the leaf. Initially empty.
+    let held_data: &mut A64Bytes<ValueSize> = &mut dummy_held_data;
+    let held_meta: &mut A8Bytes<MetaSize> = &mut dummy_held_meta;
+    // Dest represents the bucket where we will swap the held element for a new
+    // one. Floor_index corresponds to a null value.
+    let mut dest = FLOOR_INDEX;
+    let stash_index = meta_len;
+    //Look through the stash to find the element that can go the deepest, then
+    // putting it in the hold and setting dest to the target[STASH_INDEX]
+    let (_deepest_target, id_of_the_deepest_target_for_level) =
+        find_index_of_deepest_target_for_bucket::<ValueSize, Z>(stash_meta, branch.leaf, meta_len);
+    compare_and_take_held_item_if_appropriate(
+        stash_index,
+        &target_meta,
+        stash_meta,
+        stash_data,
+        id_of_the_deepest_target_for_level,
+        held_meta,
+        held_data,
+        &mut dest,
+    );
+    //Go through the branch from root to leaf, holding up to one element, swapping
+    // held blocks into destinations closer to the leaf.
+    for bucket_num in (0..meta_len).rev() {
+        //lower contains from [0..bucket_num), upper contains from [bucket_num..len).
+        // lower is the side closer to the leaf, while upper is the side closer to the
+        // root.
+        let (_, upper_data) = branch.data.split_at_mut(bucket_num);
+        let (_, upper_meta) = branch.meta.split_at_mut(bucket_num);
+
+        let bucket_data: &mut [A64Bytes<ValueSize>] = upper_data[0].as_mut_aligned_chunks();
+        let bucket_meta: &mut [A8Bytes<MetaSize>] = upper_meta[0].as_mut_aligned_chunks();
+
+        let mut dummy_to_write_data: A64Bytes<ValueSize> = Default::default();
+        let mut dummy_to_write_meta: A8Bytes<MetaSize> = Default::default();
+
+        let to_write_data: &mut A64Bytes<ValueSize> = &mut dummy_to_write_data;
+        let to_write_meta: &mut A8Bytes<MetaSize> = &mut dummy_to_write_meta;
```

Comment:
we should probably be able to replace this binding `to_write_meta` with `&mut dummy_to_write_meta`